api/server.go

```diff
@@ -719,6 +720,21 @@ func (s *server) hostsHandler(jc jape.Context) {
 	jc.Encode(hosts)
 }
 
+func (s *server) hostsNetAddressHandler(jc jape.Context) {
+	var netAddress string
+	if jc.DecodeParam("addr", &netAddress) != nil {
+		return
+	}
+	hosts, err := s.e.HostNetAddresses([]string{netAddress})
+	if jc.Check("failed to get host", err) != nil {
+		return
+	} else if len(hosts) == 0 {
+		jc.Error(ErrHostNotFound, http.StatusNotFound)
+		return
+	}
+	jc.Encode(hosts[0])
```

Comment:
Do we need a new endpoint? We could extend the one that queries hosts with this functionality.

It's also not safe to assume that this always returns a single host. If shut down my host, create a new one on the same server and then announce the same address, two hosts will have the same address in the database.

Reply:
Yeah, I guess we don't need a special endpoint.  Unlike pubkeys, net addresses aren't so unique.  With /hosts clients can just submit `{"netAddresses": ["x.x.x.x:1234", ...]}`.